### PR TITLE
zig: update to 0.13.0

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by i0ntempest** [comment]
While this builds fine, `zig version` gives:
```
0.13.0
error: Zig was built/linked incorrectly: LLVM and Clang have separate copies of libc++
       If you are dynamically linking LLVM, make sure you dynamically link libc++ too
```
I don't know if this is caused by how MP builds LLVM and Clang, but the combiler functions and I can build `ncdu` port just fine.

**Comment by pmetzger** [comment]
@i0ntempest I can't merge this if it's broken? What would you like to do next?

**Comment by i0ntempest** [comment]
> @i0ntempest I can't merge this if it's broken? What would you like to do next?

I don't know if it's broken. Changelog did not state changes in build processes and the resulting binary seems to be working fine. More likely it's triggered by how MP installs Clang and LLVM but I'm not sure. Maybe tag some C/C++ experts we have here to take a look?

**Comment by pmetzger** [comment]
@jmroot @ryandesign  Who would be good at looking at this do you think?

**Comment by ryandesign** [comment]
I don't know. Ask the developers of zig what their error message means and how they would like us to fix it.

**Comment by pmetzger** [comment]
@i0ntempest How would you like to proceed?

**Comment by i0ntempest** [comment]
I opened an upstream issue, hopefully they can provide an explanation. Meanwhile I have pushed 0.12.1.

**Comment by reneeotten** [comment]
ping - please fix or close until upstream has responded

**Comment by i0ntempest** [comment]
We had a response but the exact problem isn't clear yet. They said I "somehow have two libc++ libraries loaded into the Zig compiler process".

**Comment by sugarthc** [comment]
The issue appears to be that in 0.13.0, libClang-cpp and libLLVM are linked dynamically, where they were not in 0.12.1.  Those libraries in turn link the system libc++ library, also dynamically.  The zig executable, however, links libc++ statically and it is not clear which libc++ it is linking, but zig itself is detecting and throwing the error above.

Much seems to have changed with the build between the two releases and it is not clear to me where these changes are being introduced.  I'm not well versed in cmake and I certainly have next to no understanding of the zig build system.  The interaction between the two seems rather nightmarish to me.

It seems to be the confluence of changes in the llvm/clang cmake files and very minor changes in zig build files.  I wish I could tell which changes but at this point I cannot.  I will keep looking, though.

The bottom line (I think) is that either llvm and clang need to be statically linked as they were in 0.12.1 (which I think triggers a change in which libc++ is linked) or the stage3 zig binary needs to link the system libc++ dynamically.  I don't know how to effect either of these changes at the moment.

The net effect of the former is that zig goes back to being an approximately 180MB binary with the macports llvm/clang as build-only dependencies or zig is a 20-ish MB binary that has macports llvm/clang as runtime dependencies.

**Comment by sugarthc** [comment]
I poked at this a bit more and have found that adding the following to the Portfile appears to reproduce what what happening with the 0.12.1 port.  That is to say that the zig binary statically links llvm/clang and (some) libc++.  I'm not sure if it's the system one (probably not) or the one in llvm or zig's own.  In any case this produces a 184MB zig binary that no longer emits the error above.

configure.args-append -DZIG_STATIC_LLVM=ON

I still have no idea why this is needed to reproduce the default behavior from the previous release.  I've spotted no diffs in either llvm/clang or zig that points specifically to this.  I'm guessing something to do with library detection in the new cmake files in llvm/clang.   If somebody discovers this I'd love to hear about it.

**Comment by sugarthc** [comment]
I’d also like to reiterate that neither the previous version nor this one with this change applied requires llvm or clang as dependencies unless you are building from source, so it’d be nice if the clang dependency could be moved to depends_build

**Comment by sugarthc** [comment]
To this point I can see no clear path forward for dynamic linking of llvm/clang AND zig itself.  I found this [upstream PR](https://github.com/ziglang/zig/pull/13498) that seems to want to accomplish this but it was closed without merging. Apparently because nobody could be bothered to rebase the patch.

**Comment by carlocab** [comment]
Should be fixed upstream in https://github.com/ziglang/zig/pull/23264.
